.github/workflows/build_pr_container.yaml

```diff
@@ -0,0 +1,38 @@
+name: build PR container
+on:
+  pull_request:
+    paths:
+      - '*.go'
+      - './Dockerfile'
+
+jobs:
+  build:
+    name: Build PR image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Log in to the Github Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build container
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: |
+            ghcr.io/${{ github.repository }}:PR${{ github.event.number }}
+            ghcr.io/${{ github.repository }}:sha-${{ github.sha }}
```

Comment:
Why do we need two tags for the same image? Is that for when `{{ github.event.number }}` returns an empty string?

Reply:
A forced push will change the `sha` but not the PR number

Reply:
Good point.